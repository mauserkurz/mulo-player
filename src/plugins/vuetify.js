import Vue from 'vue';
import Vuetify from 'vuetify';
import { THEME as theme } from '@/constants';

Vue.use(Vuetify);

export default new Vuetify({ theme });
