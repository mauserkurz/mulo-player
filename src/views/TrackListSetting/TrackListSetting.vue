<template>
  <v-row class="track-list-setting">
    <v-col cols="4">
      <v-select
        :value="filterType"
        :items="filterTypeList"
        item-text="text"
        item-value="type"
        @input="onFilterTypeInput"/>
    </v-col>

    <v-col cols="4">
      <v-select
        :value="sortType"
        :items="sortTypeList"
        item-text="text"
        item-value="type"
        @input="onSortTypeInput"/>
    </v-col>
  </v-row>
</template>

<script>
import { SORT_TRACK_TYPE_LIST, FILTER_TRACK_TYPE_LIST } from '@/constants';

export default {
  name: 'TrackListSetting',

  props: {
    sortType: {
      type: String,
      required: true,
    },

    filterType: {
      type: String,
      required: true,
    },
  },

  data() {
    return {
      sortTypeList: SORT_TRACK_TYPE_LIST,
      filterTypeList: FILTER_TRACK_TYPE_LIST,
    };
  },

  methods: {
    onSortTypeInput(value) {
      this.$emit('input-sort-type', value);
    },

    onFilterTypeInput(value) {
      this.$emit('input-filter-type', value);
    },
  },
};
</script>

<style scoped lang="less">
.track-list-setting {}
</style>
