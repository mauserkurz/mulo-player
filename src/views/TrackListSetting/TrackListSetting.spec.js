import { mount, shallowMount, createLocalVue } from '@vue/test-utils';
import vuetify from '@/plugins/vuetify';
import { FILTER_TRACK_TYPE_KEYS, SORT_TRACK_TYPE_KEYS } from '@/constants';
import TrackListSetting from './TrackListSetting.vue';

const localVue = createLocalVue();

describe('Component TrackListSetting', () => {
  const defaultProps = {
    sortType: SORT_TRACK_TYPE_KEYS.ABC,
    filterType: FILTER_TRACK_TYPE_KEYS.TEXT,
  };
  const createWrapper = ({ isWithoutStubs = false, propsData = {}, ...options } = {}) => {
    const renderer = isWithoutStubs ? mount : shallowMount;
    const wrapper = renderer(TrackListSetting, {
      localVue,
      vuetify,
      propsData: Object.assign({}, defaultProps, propsData),
      ...options,
    });

    return { wrapper };
  };

  describe('snapshots', () => {
    it('should match snapshot', () => {
      const { wrapper } = createWrapper({ isWithoutStubs: true });

      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('events', () => {
    it('should after select filterType emit event input-filter-type with value', () => {
      const { wrapper } = createWrapper({ isWithoutStubs: true });

      wrapper.find('.v-select').vm
        .$emit('input', FILTER_TRACK_TYPE_KEYS.NEW);
      expect(wrapper.emitted()['input-filter-type']).toEqual([[FILTER_TRACK_TYPE_KEYS.NEW]]);
    });

    it('should after select sortType emit event input-sort-type with value', () => {
      const { wrapper } = createWrapper({ isWithoutStubs: true });

      wrapper.findAll('.v-select').at(1).vm
        .$emit('input', SORT_TRACK_TYPE_KEYS.CBA);
      expect(wrapper.emitted()['input-sort-type']).toEqual([[SORT_TRACK_TYPE_KEYS.CBA]]);
    });
  });
});
