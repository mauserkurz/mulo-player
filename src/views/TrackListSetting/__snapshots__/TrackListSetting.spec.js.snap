// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component TrackListSetting snapshots should match snapshot 1`] = `
<div class="row track-list-setting">
  <div class="col col-4">
    <div class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field v-select">
      <div class="v-input__control">
        <div role="button" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-3" class="v-input__slot">
          <div class="v-select__slot">
            <div class="v-select__selections">
              <div class="v-select__selection v-select__selection--comma">Search name</div><input id="input-3" readonly="readonly" type="text" aria-readonly="true" autocomplete="off">
            </div>
            <div class="v-input__append-inner">
              <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
            </div><input type="hidden" value="text">
          </div>
          <div class="v-menu">
            <div class="v-menu__content theme--light " style="max-height: 304px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"></div>
          </div>
        </div>
        <div class="v-text-field__details">
          <div class="v-messages theme--light"><span name="message-transition" tag="div" class="v-messages__wrapper"></span></div>
        </div>
      </div>
    </div>
  </div>
  <div class="col col-4">
    <div class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field v-select">
      <div class="v-input__control">
        <div role="button" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-10" class="v-input__slot">
          <div class="v-select__slot">
            <div class="v-select__selections">
              <div class="v-select__selection v-select__selection--comma">From A to Z</div><input id="input-10" readonly="readonly" type="text" aria-readonly="true" autocomplete="off">
            </div>
            <div class="v-input__append-inner">
              <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
            </div><input type="hidden" value="abc">
          </div>
          <div class="v-menu">
            <div class="v-menu__content theme--light " style="max-height: 304px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"></div>
          </div>
        </div>
        <div class="v-text-field__details">
          <div class="v-messages theme--light"><span name="message-transition" tag="div" class="v-messages__wrapper"></span></div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
